great_expectations/data_context/data_context/abstract_data_context.py

```diff
@@ -1201,7 +1208,9 @@ def delete_datasource(
             raise ValueError(f"Datasource {datasource_name} not found")
 
         if save_changes:
-            datasource_config = datasourceConfigSchema.load(datasource.config)
+            datasource_config = datasourceConfigSchema.load(
```

Comment:
I was trying to avoid adding extra conditionals.
I do anticipate I will be adding one here or somehow making it unnecessary in a follow-up to fix the `save` logic.